neuraxle/pipeline.py

```diff
@@ -394,34 +396,58 @@ def _load_checkpoint(self, data_container: DataContainer) -> Tuple[NamedTupleLis
 
         return self.steps_as_tuple[new_starting_step_index:], starting_step_data_container
 
-    def _try_loading_pipeline_checkpoint(self, pipeline: 'Pipeline', new_starting_step_index) -> bool:
+    def _try_loading_pipeline_checkpoint(
+            self,
+            starting_step_data_container: DataContainer,
+            new_starting_step_index
+    ) -> bool:
         """
-        Load persisted pipeline if the bytecode of the steps before
-        the latest checkpoint have not changed
+        Load persisted pipeline steps before checkpoint
+        if the steps before the latest checkpoint have not changed
 
-        :param pipeline: loaded cached pipeline
+        :param starting_step_data_container: loaded cached pipeline
         :param new_starting_step_index:
         :return: true if loading succeeded
         """
+        cached_pipeline = self.pipeline_repository.load(self, starting_step_data_container)
```

Comment:
why is this isn't saved anywhere? I don't understand

Reply:
Oh, I realized that perhaps it was a side effect done in`self._load_cached_pipeline_steps`. Not that bad, but better might be possible.

Reply:
it is saved only when we want to save it : when we save a checkpoint step...